scripts/ci/run_perftest.sh

```diff
@@ -119,125 +139,153 @@ force_compaction() {
+    fi
   done
 
   # restart daemon
   systemctl unmask influxdb.service
   systemctl start influxdb
 }
 
-# Run and record tests
-
-# General ingest and query tests
-for scale in 50 100 500; do
-  # generate bulk data
-  scale_string="scalevar-$scale"
-  scale_seed_string="$scale_string-seed-$seed"
-  data_fname="influx-bulk-records-usecase-devops-$scale_seed_string.txt"
-  $GOPATH/bin/bulk_data_gen --seed=$seed --use-case=devops --scale-var=$scale --format=influx-bulk > ${DATASET_DIR}/$data_fname
-  query_files=""
-  for format in http flux-http; do
-    query_fname="query-devops-$scale_seed_string-$format.txt"
-    $GOPATH/bin/bulk_query_gen --seed=$seed --use-case=devops --scale-var=$scale --format=influx-$format --db $db_name --queries 10000 --query-type 8-host-1-hr > ${DATASET_DIR}/$query_fname
-    query_files="$query_files $query_fname"
-  done
-
-  # Test loop
-  for parseme in "5000:2" "5000:20" "15000:2" "15000:20"; do
-    batch=$(echo $parseme | cut -d: -f1)
-    workers=$(echo $parseme | cut -d: -f2)
-
-    # generate load data
-    load_opts="-batch-size=$batch -workers=$workers -urls=http://${NGINX_HOST}:8086 -do-abort-on-exist=false -do-db-create=true -backoff=1s -backoff-timeout=300m0s"
-    if [ -z $INFLUXDB2 ] || [ $INFLUXDB2 = true ]; then
-      load_opts="$load_opts -organization=$TEST_ORG -token=$TEST_TOKEN"
-    fi
-
-    # run ingest tests
-    cat ${DATASET_DIR}/$data_fname | $GOPATH/bin/bulk_load_influx $load_opts | jq ". += {branch: \"${INFLUXDB_VERSION}\", commit: \"${TEST_COMMIT}\", time: \"$datestring\", i_type: \"${DATA_I_TYPE}\"}" > $working_dir/test-ingest-$scale_string-batchsize-$batch-workers-$workers.json
-
-    force_compaction
-
-    # run influxql and flux query tests
-    for query_file in $query_files; do
-      format=$(echo $query_file | cut -d '-' -f7)
-      format=${format%.txt}
-      cat ${DATASET_DIR}/$query_file | ${GOPATH}/bin/query_benchmarker_influxdb --urls=http://localhost:8086 --debug=0 --print-interval=0 --workers=$workers --json=true --organization=$TEST_ORG --token=$TEST_TOKEN | jq ". += {branch: \"${INFLUXDB_VERSION}\", commit: \"${TEST_COMMIT}\", time: \"$datestring\", i_type: \"${DATA_I_TYPE}\", query_format: \"$format\"}" > $working_dir/test-query-$scale_string-format-$format-workers-$workers.json
-    done
-
-    clean_db
-  done
-done
-
-# Metaquery tests
-
-# Controls how many data points will be generated. The total number of points
-# will be scale_var * scale_var, where each point is (approximately) a unique
-# series.
-scale_var=1000
-
-# How many queries to generate.
-queries=100
-
 # The time range controls both the span over which data is generated, and the
 # span over which queries will be performed. timestamp-start and timestamp-end
 # must be provided to both the data generation and query generation commands
 # and must be the same to ensure that the queries cover the data range.
-start="2019-01-01T00:00:00Z"
-end="2020-01-01T00:00:00Z"
+start_time() {
```

Comment:
I bet we could use the same start and stop time for all the queries, including the metaqueries. This would mix the metaquery data in the same time window as the IoT data but I doubt that would make much difference. I think you can keep it like it is here but I'll probably try that in the future if you haven't already.

Reply:
My first attempt did use your metaquery start/end times for the iot dataset, but generating a year's worth of data from that use case produced way too much output.